src/core/internal/dassert.d

```diff
@@ -152,6 +152,20 @@ private string miniFormat(V)(const scope ref V v)
             return miniFormat(*cast(T*) &v);
         }
     }
+    // Format enum members using their name
+    else static if (is(V BaseType == enum))
+    {
+        // Always generate repeated if's instead of switch to skip the detection
+        // of non-integral enums. This method doesn't need to be fast.
+        static foreach (mem; __traits(allMembers, V))
+        {
+            if (v == __traits(getMember, V, mem))
+                return mem;
+        }
+
+        // Format invalid enum values as their base type
+        return "cast(" ~ V.stringof ~ ") " ~ miniFormat(*(cast(BaseType*) &v));
```

Comment:
Doesn't this allocate ? Also `miniFormat!BaseType(v)` should work, no ?

Reply:
> Doesn't this allocate ?

Yes, but string concatenation is used in several places. But I could also re-use `combine` here...

>  Also `miniFormat!BaseType(v)` should work, no ?

No because `miniFormat`'s accepts it's parameter by `ref`

Reply:
> Yes, but string concatenation is used in several places.

If we are to make this the default, `assert(a != b)` should not allocate.
Note that it currently does anyway, because of `throw new AssertError`, but it really shouldn't, because that makes it unusable in destructors.

Reply:
Guess this could use `combine` (or some other utility method) s.t. we can change the allocation strategy in the future.

But that will always be a problem for a user-defined `toString()`s that are not `@nogc` (unless those are skipped while running destructors)

Reply:
Assert errors shouldn't really be caught. Can we just malloc it?

Reply:
Another solution -- set up a "for asserts" allocation that uses malloc, and then frees everything just before the program exits.

Reply:
> But that will always be a problem for a user-defined `toString()`s that are not `@nogc` (unless those are skipped while running destructors)

that's going to be a really tough problem to solve.

Reply:
> Can we just malloc it?

hah, just looked at what combine does, and it does just that ;) That's the solution right there.

Reply:
Done. Maybe we should rework `miniFormat` to use an `OutputRange`-like buffer s.t. it doesn't need to reallocate to build the final string.